sklearn/metrics/_pairwise_distances_reduction/_base.pxd.tp

```diff
@@ -1,23 +1,40 @@
+{{py:
+
+implementation_specific_values = [
+    # Values are the following ones:
+    #
+    #       name_suffix, INPUT_DTYPE_t, INPUT_DTYPE
+    #
+    # We also use 64bit types as defined in `sklearn.utils._typedefs`
+    # to maintain consistency.
```

Comment:
```suggestion
    # We also use the float64 dtype and C-type names as defined in
    # `sklearn.utils._typedefs` to maintain consistency.
```

Reply:
Actually, wouldn't that be a good opportunity to stop using the `DTYPE(_t)` names that are quite confusing? Better be explicit.

Reply:
I agree that those aren't proper and understandable names, yet I would rather have other PRs unify custom types definitions and names. What do you think?